nmigen/lib/io.py

```diff
@@ -1,27 +1,18 @@
 from .. import *
+from ..hdl.rec import *
 
 
-__all__ = ["TSTriple", "Tristate"]
+__all__ = ["TripleLayout", "Tristate"]
 
 
-class TSTriple:
-    def __init__(self, shape=None, min=None, max=None, reset_o=0, reset_oe=0, reset_i=0,
-                 name=None):
-        self.o  = Signal(shape, min=min, max=max, reset=reset_o,
-                         name=None if name is None else name + "_o")
-        self.oe = Signal(reset=reset_oe,
-                         name=None if name is None else name + "_oe")
-        self.i  = Signal(shape, min=min, max=max, reset=reset_i,
-                         name=None if name is None else name + "_i")
-
-    def __len__(self):
-        return len(self.o)
-
-    def elaborate(self, platform):
-        return Fragment()
-
-    def get_tristate(self, io):
-        return Tristate(self, io)
+class TripleLayout(Layout):
+    def __init__(self, shape):
+        fields = [
+            ("o",  shape, DIR_FANOUT),
+            ("oe", 1,     DIR_FANOUT),
+            ("i",  shape, DIR_FANIN)
+        ]
+        super().__init__(fields)
```

Comment:
I think this can be just a function `def TripleLayout`, since subclassing adds no value here.